server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Rights.scala

```diff
@@ -0,0 +1,147 @@
+/** *************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.mail
+
+import org.apache.james.core.Username
+import org.apache.james.mailbox.model.MailboxACL
+import org.apache.james.mailbox.model.MailboxACL.{EntryKey, Rfc4314Rights}
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+object Right {
+  val UNSUPPORTED: Option[Boolean] = None
```

Comment:
I'm not fan of it as :
 - Enumeration.Value don't allow to pass arbitrary members
 - This means that `toMailboxRight` property would be migrated to an `object` behaviour instead of class

Reply:
I did not understand your answer.